inst/update_deployer.R

```diff
@@ -0,0 +1,41 @@
+#' Update a package in a deployer from the source code
+#'
+#' @param pkg the path to a package source
+#' @param repo path to a drat repo. Defaults to both the current and 'drat' repo 
+#' @param internet a future parameter to update packages from internet if it is
+#'   available.
+#' @examples
+#'
+#' deploy_package("../linelist") # update the linelist package, assuming its
+#'   source lives one directory upstream of this deployer
+deploy_package <- function(pkg = NULL, repo = c(here::here(), here::here("drat")), internet = FALSE) {
+  if (is.null(pkg)) {
+    msg <- paste(
+      "Please supply a path to a package source"
+    )
+    stop(msg)
+  }
+  stopifnot(file.exists(pkg))
+  
+  vers <- read.dcf(file.path(pkg, "DESCRIPTION"))[, "Version"]
+  message("Building source package ...")
+  src <- devtools::build(pkg, path = here::here(), binary = FALSE)
+  message(sprintf("Building binary package for %s", R.version$os))
+  bin <- devtools::build(pkg, path = here::here(), binary = TRUE)
+  
+
+  for (i in repo) {
```

Comment:
I'm confused about the two `repo` being passed in here and the loop, but presume that this is mostly harmless? I don't really understand the need for `here` but I'm probably being thick.

Reply:
I noticed several weeks ago that the packages are stored in either the top-level directory or in a drat directory (but activate.R was only set up to consider the top-level repo). I honestly don't know which one is favored and I'm a bit paranoid about what would happen if a package is updated on one repo and not the other. 

I used `here` to try and avoid issues people were having earlier with not being in the right working directory when running activate.R (See #14). Thinking about this more, it doesn't really help much if they are in the wrong working directory. I can probably add a catch to look for the known files in this directory.

Reply:
The two repos - the `drat` one is for the two stage calculation of dependencies - we organise non-github repos to come down into a mini-CRAN like repo and then use that to generate the main one.  you could delete entirely the drat one if you wanted.

If it was the case that it was only packages already in the mini drat repo that were being updated then there are tools in `provisionr` for refreshing those packages, but I'm not sure with the original problem description if that would be a good fit or not

Reply:
> The two repos - the drat one is for the two stage calculation of dependencies - we organise non-github repos to come down into a mini-CRAN like repo and then use that to generate the main one. you could delete entirely the drat one if you wanted.

I did not realise that. I think I ran into an issue a few weeks ago where I couldn't install a package that I knew was on the deployer and it was fixed by adding the drat folder as a repo. Now that I compare them, it seems that there's no reason to include it at all.

I think, at minimum, what needs to happen is basically encapsulated in `drat:::insert()`: shove the tarball in, archive the old version (if needed), and update the packages list.